docs/data_sharing_guide.md

```diff
@@ -0,0 +1,338 @@
+credentials = get_minio_credentials()
+print(f"Username: {credentials.username}")
+print(f"Access key: {credentials.access_key}")
+print(f"Secret key: {credentials.secret_key}")
+
+# Get your SQL warehouse prefix
+sql_warehouse = get_my_sql_warehouse()
+print(f"SQL warehouse prefix: {sql_warehouse.sql_warehouse_prefix}")
+
+# The pre-initialized governance client is also available
+print(f"Governance client ready: {governance is not None}")
+```
+
+### Workspace and Policy Information
+
+```python
+# Get comprehensive workspace information
+workspace = get_my_workspace()
+print(f"Username: {workspace.username}")
+print(f"Home paths: {workspace.home_paths}")
+print(f"Groups: {workspace.groups}")
+print(f"Total accessible paths: {len(workspace.accessible_paths)}")
+
+# Get detailed policy information
+policies = get_my_policies()
+print(f"Home policy: {policies.user_home_policy.policy_name}")
+print(f"System policy: {policies.user_system_policy.policy_name}")
+print(f"Group policies: {len(policies.group_policies)}")
+```
+
+## Sharing SQL Warehouse Tables
+
+### Share Tables with Users
+
+```python
+# Share a table with specific users
+response = share_table(
+    namespace="research",
+    table_name="climate_data",
+    with_users=["bob", "alice"]  # kbase usernames
+)
+
+print(f"Shared with users: {response.shared_with_users}")
+print(f"Success count: {response.success_count}")
+if response.errors:
+    print(f"Errors: {response.errors}")
+```
+
+### Share Tables with Groups
+
+```python
+# Share a table with groups
+response = share_table(
+    namespace="analytics", 
+    table_name="user_metrics",
+    with_groups=["data-science-team", "research-group"]
+)
+
+print(f"Shared with groups: {response.shared_with_groups}")
+```
+
+### Share with Both Users and Groups
+
+```python
+# Share with combination of users and groups
+response = share_table(
+    namespace="experiments",
+    table_name="results_2024",
+    with_users=["researcher1", "analyst2"],
+    with_groups=["biology-lab"]
+)
+
+print(f"Successfully shared with {response.success_count} recipients")
+```
+
+## Unsharing SQL Warehouse Tables
+
+### Remove Access from Users
+
+```python
+# Remove access from specific users
+response = unshare_table(
+    namespace="research",
+    table_name="climate_data",
+    from_users=["bob"]  # Remove bob's access, alice keeps access
+)
+
+print(f"Removed access from: {response.unshared_from_users}")
+```
+
+### Remove Access from Groups
+
+```python
+# Remove group access
+response = unshare_table(
+    namespace="analytics",
+    table_name="user_metrics", 
+    from_groups=["data-science-team"]
+)
+
+print(f"Removed group access from: {response.unshared_from_groups}")
+```
+
+### Unshare from Both Users and Groups
+
+```python
+# Unshare from both users and groups
+response = unshare_table(
+    namespace="sensitive",
+    table_name="confidential_data",
+    from_users=["bob", "alice"],
+    from_groups=["research-team", "data-scientists"]
+)
```

Comment:
In the future a `completely_unshare` boolean could be useful to remove all users and / or groups